homeassistant/components/sensor/seventeentrack.py

```diff
@@ -0,0 +1,274 @@
+    """Define a summary sensor."""
+
+    def __init__(self, data, status, initial_state):
+        """Initialize."""
+        self._attrs = {ATTR_ATTRIBUTION: DEFAULT_ATTRIBUTION}
+        self._data = data
+        self._state = initial_state
+        self._status = status
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        return self._attrs
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return 'mdi:package'
+
+    @property
+    def name(self):
+        """Return the name."""
+        return 'Packages {0}'.format(self._status)
+
+    @property
+    def state(self):
+        """Return the state."""
+        return self._state
+
+    @property
+    def unique_id(self):
+        """Return a unique, HASS-friendly identifier for this entity."""
+        return 'summary_{0}_{1}'.format(
+            self._data.account_id, slugify(self._status))
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return 'packages'
+
+    async def async_update(self):
+        """Update the sensor."""
+        await self._data.async_update()
+
+        self._state = self._data.summary[self._status]
+
+
+class SeventeenTrackPackageSensor(Entity):
+    """Define an individual package sensor."""
+
+    def __init__(self, data, package):
+        """Initialize."""
+        self._attrs = {
+            ATTR_ATTRIBUTION: DEFAULT_ATTRIBUTION,
+            ATTR_DESTINATION_COUNTRY: package.destination_country,
+            ATTR_INFO_TEXT: package.info_text,
+            ATTR_LOCATION: package.location,
+            ATTR_ORIGIN_COUNTRY: package.origin_country,
+            ATTR_PACKAGE_TYPE: package.package_type,
+            ATTR_TRACKING_INFO_LANGUAGE: package.tracking_info_language,
+        }
+        self._data = data
+        self._state = package.status
+        self._tracking_number = package.tracking_number
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        return self._attrs
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return 'mdi:package'
+
+    @property
+    def name(self):
+        """Return the name."""
+        return self._tracking_number
+
+    @property
+    def state(self):
+        """Return the state."""
+        return self._state
+
+    @property
+    def unique_id(self):
+        """Return a unique, HASS-friendly identifier for this entity."""
+        return 'package_{0}_{1}'.format(
+            self._data.account_id, self._tracking_number)
+
+    async def async_update(self):
+        """Update the sensor."""
+        await self._data.async_update()
+
+        try:
+            package = next(
+                iter([
+                    p for p in self._data.packages
+                    if p.tracking_number == self._tracking_number
+                ]))
+        except StopIteration:
```

Comment:
If there's an error with the data update, `self.packages` is set to empty list and all entities will be removed. I think we shouldn't do that but keep current entities in that case and mark them as unavailable, until the data update succeeds again.

Reply:
I went about this slightly differently in 61a19d7 – let me know what you think (and if I missed something).

Reply:
Now it looks like upon error we just continue reporting the same state as before. It doesn't look like the entities will be marked as unavailable. Am I missing anything?

Reply:
Ah, great point. Solved one problem, but left the other. Let me rethink.